internal/js_library/js_library.bzl

```diff
@@ -285,9 +285,10 @@ def _impl(ctx):
             deps = ctx.attr.deps,
         ))
         providers.append(OutputGroupInfo(types = decls))
-    elif has_directories:
-        # If this is directory artifacts npm package then always provide declaration_info
-        # since we can't scan through files
+    else:
+
+        # Always propagate declaration_info so that downstream ts_project can depend on js_library
+        # Also important when has_directories since we can't scan through files
```

Comment:
@alexeagle reopening. This change seems to fix the issue of ts_project dependending on js_library